llama_index/tools/tool_spec/base.py

```diff
@@ -0,0 +1,33 @@
+"""Base tool spec class."""
+
+from abc import ABC, abstractmethod
+from typing import List, Optional
+from llama_index.tools.types import ToolMetadata
+from llama_index.tools.function_tool import FunctionTool
+from inspect import signature
+
+
+class BaseToolSpec:
+    """Base tool spec class."""
+
+    # list of functions that you'd want to convert to spec
+    spec_functions: List[str]
+
+    def to_tool_list(
+        self,
+        func_to_metadata_mapping: Optional[ToolMetadata] = None,
+    ) -> List[FunctionTool]:
+        """Convert tool spec to list of tools."""
+        func_to_metadata_mapping = func_to_metadata_mapping or {}
+        tool_list = []
+        for func_name in self.spec_functions:
+            func = getattr(self, func_name)
+            metadata = func_to_metadata_mapping.get(func_name, None)
+            if metadata is None:
+                name = func_name
+                docstring = func.__doc__
```

Comment:
Might need a good way to communicate to the user that the docstring is required, as well as how the description is built (since the tool description can be pretty important). Maybe a docs page eventually?